cocos/base/CachedArray.h

```diff
@@ -34,17 +34,17 @@ template <typename T>
 class CachedArray : public Object {
 public:
     explicit CachedArray(uint size = 1U) {
-        _size = 0;
-        _capacity = std::max(size, 1u);
-        _array = CC_NEW_ARRAY(T, _capacity);
+        _size     = 0;
+        _capacity = std::max(size, 1U);
+        _array    = CC_NEW_ARRAY(T, _capacity);
     }
 
     // The rule of five applies here
     ~CachedArray() override {
         CC_SAFE_DELETE_ARRAY(_array);
     }
 
-    CachedArray(const CachedArray &other)
+    CachedArray(const CachedArray &other) noexcept
```

Comment:
Why need noexcept?

Reply:
It is beneficial to declare all four copy/move constructors/assignment operators `noexcept`, as we have discussed earlier.

Reply:
as i remembered, only move constructor/assignment operators will add noexcept.

Reply:
Oh okay, but still I would suggest we add `noexcept` for all four of them, because it is trivial to do and generally a more consistent coding style. Also there are checks like `std::is_nothrow_copy_constructible`, etc. detects exactly this information. If the move constructor/asssignment operator is important for reallocations through the use of `std::is_nothrow_move_constructible`, then there may exist some more scenarios where the `noexcept` status of the copy counterparts are beneficial too.

Reply:
As we discussed, adding noexcept is just for performance, if a function is good for it, we can add it. And move constructor/assignment operators we know are always benefit for noexcept.

Of course we can create a new issue and discuss the rule again if you think it is not good. I think we should spend more time to think about it.

I think we should obey current rule until we decide to change it.

Reply:
alright